app/lib/actions/proxy_action.rb

```diff
@@ -218,10 +218,8 @@ def start_or_resume
     end
 
     def get_proxy_data(response)
-      proxy_data = response['actions'].detect do |action|
-        action['class'] == proxy_action_name || action.fetch('input', {})['proxy_operation_name'] == proxy_operation_name
-      end
-      proxy_data.fetch('output', {})
+      response['actions'].detect { |action| action['input']['task_id'] == task.id }
```

Comment:
given how `action.fetch('input', {})` was used defensively before, are we sure action['input'] is always a hash? or should we also write that more defensively?

Reply:
If there's an action there should be an input and if there's an input it is a hash. Otoh playing it safe is almost free so I can do it that way

Reply:
Fixed